src/components/frame/Extensions/pages/MultiSigPage/hooks.ts

```diff
@@ -0,0 +1,161 @@
+
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import { ExtensionSaveSettingsContext } from '~context/ExtensionSaveSettingsContext/ExtensionSaveSettingsContext.ts';
+import { ActionTypes } from '~redux/actionTypes.ts';
+import { isInstalledExtensionData } from '~utils/extensions.ts';
+
+import {
+  MultiSigThresholdType,
+  type DomainThresholdConfig,
+  type MultiSigSetupFormValues,
+} from './types.ts';
+import {
+  getDomainThresholds,
+  getInitialDomainConfig,
+  getInitialGlobalThreshold,
+  getThresholdType,
+} from './utils.ts';
+
+export const useBackNavigation = ({ extensionData }) => {
+  const navigate = useNavigate();
+
+  /*
+   * If we arrive here but the extension is not installed go back to main extension details page
+   */
+  useEffect(() => {
+    if (!isInstalledExtensionData(extensionData)) {
+      navigate(-1);
+    }
+  }, [extensionData, navigate]);
+};
+
+export const useThresholdData = ({ extensionData }) => {
+  const multiSigConfig = extensionData.params?.multiSig || null;
+
+  const {
+    colony: { colonyAddress, domains },
+  } = useColonyContext();
+
+  const {
+    register,
+    setValue,
+    getValues,
+    trigger,
+    formState: { errors, isValid },
+    reset,
+  } = useForm<MultiSigSetupFormValues>({
+    defaultValues: {
+      globalThreshold: getInitialGlobalThreshold(multiSigConfig),
+    },
+    mode: 'onChange',
+  });
+
+  const { handleIsVisible, handleSetActionType, resetAll, callback } =
+    useContext(ExtensionSaveSettingsContext);
+
+  const [thresholdType, setThresholdType] = useState<MultiSigThresholdType>(
+    getThresholdType(multiSigConfig?.colonyThreshold),
+  );
+
+  const [domainThresholdConfigs, setDomainThresholdConfigs] = useState<
+    DomainThresholdConfig[]
+  >([]);
+
+  useEffect(() => {
+    handleIsVisible(true);
+    handleSetActionType(ActionTypes.MULTISIG_SET_THRESHOLDS);
+
+    return () => {
+      resetAll();
+      reset();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    setValue('colonyAddress', colonyAddress);
+  }, [colonyAddress, setValue]);
+
+  useEffect(() => {
+    if (!domains || !domains.items || !multiSigConfig) {
+      return;
+    }
+
+    const domainsExcludingRoot = domains.items.filter(
+      (domain): domain is NonNullable<typeof domain> =>
+        domain !== null && !domain.isRoot,
+    );
+
+    const tempDomainThresholdConfigs = domainsExcludingRoot.map((domain) => {
+      const { threshold, ...rest } = getInitialDomainConfig(
+        domain,
+        multiSigConfig,
+      );
+
+      setValue(rest.name, threshold);
+
+      return rest;
+    });
+
+    setDomainThresholdConfigs(tempDomainThresholdConfigs);
+  }, [domains, multiSigConfig, setValue]);
+
+  useImperativeHandle(callback, () => ({
+    getValues: async () => {
+      await trigger(undefined, { shouldFocus: true });
+      if (!isValid) throw new Error('Error within form');
```

Comment:
No need to use await since you’re not making use of the returned value. But looking this flow, it seems your intention is to validate all fields and react right away right? Maybe you want to assign a value in this case and make use of it so like:

```js
const isValid = await trigger(undefined, { shouldFocus: true }); // isValid is false if there’s at least one error in the form

// Then use it however you please, your majesty
if (!isValid) {
  // foobar
}
```